### PR TITLE
basic support for casa sync

### PR DESCRIPTION
**Pull request by gdestuynder**
FYI I run into a 500 error on the first project, so diagnosing this with biztera before we merge this into something that deploys.
I also plan to add a lookup for the bugzilla user email and *attempt* to mirror it in biztera. if it fails, it would just leave whatever was set, so this is for convenience only (so that you get notification in biztera if people participate there)

### CODE REVIEW

**Comment by gdestuynder** [comment]
Fixed, it ran once also.

Note: we probably should decide what to do for INVALID and DUPLICATE. RIght now it leaves as "pending" on casa. We could choose "completed" though its not exactly right.

**Comment by gdestuynder** [comment]
Last commit adds auto assignement, which i also ran once

**Comment by gdestuynder** [comment]
Run sample:

```
127 kang@xps13 ..fosec/infosec-risk-management-bugzilla (git)-[master] % ./assigner.py -d --dry-run                                                                                                  :(
[15:39:25] INFO [__main__.autoassign:207] No unassigned bugs for component
[15:39:26] INFO [__main__.autoassign:207] No unassigned bugs for component
[15:39:26] DEBUG [__main__.autocasa:89] Analyzing 10 closed bugs...
[15:39:28] WARNING [__main__.autocasa:113] Project REDACTED already has a security status set (done), skipping!
[15:39:31] INFO [__main__.autocasa:136] Assignee/delegator in CASA is already correct, no changes made REDACTED  (REDACTED@mozilla.com) for project REDACTED)
[15:39:31] INFO [__main__.autocasa:145] Would attempt to set status INVALID on project https://biztera.com/projects/REDACTEDfor bug https://bugzilla.mozilla.org/REDACTED (dry run prevented this)
[15:39:33] WARNING [__main__.autocasa:113] Project REDACTED already has a security status set (done), skipping!
[15:39:35] WARNING [__main__.autocasa:113] Project REDACTED already has a security status set (done), skipping!
[15:39:37] WARNING [__main__.autocasa:113] Project REDACTED already has a security status set (done), skipping!
[15:39:39] WARNING [__main__.autocasa:113] Project REDACTED already has a security status set (done), skipping!
[15:39:40] WARNING [__main__.autocasa:126] Could not match Bugzilla assignee: nobody@mozilla.org with Casa, using defaults
[15:39:40] INFO [__main__.autocasa:145] Would attempt to set status DUPLICATE on project https://biztera.com/projects/REDACTED for bug https://bugzilla.mozilla.org/REDACTED (dry run prevented this)
[15:39:42] INFO [__main__.autocasa:136] Assignee/delegator in CASA is already correct, no changes made REDACTED (REDACTED@mozilla.com) for project REDACTED)
[15:39:42] INFO [__main__.autocasa:145] Would attempt to set status INVALID on project https://biztera.com/projects/REDACTED for bug https://bugzilla.mozilla.org/REDACTED (dry run prevented this)
[15:39:43] WARNING [__main__.autocasa:113] Project REDACTED already has a security status set (done), skipping!
[15:39:46] INFO [__main__.autocasa:133] Setting new assignee/delegator in CASA to REDACTED (REDACTED@mozilla.com) for project REDACTED
[15:39:46] INFO [__main__.autocasa:145] Would attempt to set status INVALID on project https://biztera.com/projects/REDACTED for bug https://bugzilla.mozilla.org/REDACTED (dry run prevented this)
[15:39:46] DEBUG [__main__.autocasa:147] Casa analysis completed
```

File: assigner.py [hunk 1]
```diff
@@ -35,24 +37,119 @@ def main():
...
+
+        # Some basic checks that we can update that project
+        ## Have bugzilla support?
+        casa_project = capi.casa_get_project(casa_data.get('project_id'))
+        if casa_project['syncedToIntegrations']['bugzilla'] is not True:
+            logger.warning('Project {} has no bugzilla integration, skipping!'.format(project_id))
+            continue
+
+        # Is already approved/disapproved in some way?
+        ## XXX This means Bugzilla cannot override a status already set, thus, if you set "WONTFIX" in bugzilla,
```
Comment (claudijd) [inline assigner.py]: Was there intent to fill something in for "XXX"?

File: casa.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+        See also `decision_map` below
+        """
+        # Map decision bugzilla=>(casa decision label, casa decision)
+        decision_map = {
+                         'FIXED': ('Completed: No oustanding issues found', 'done'),
+                         'INVALID': ('Pending', 'none'),
+                         'DUPLICATE': ('Pending', 'none'),
+                         'WONTFIX': ('Do not use', 'rejected'),
+                         'INCOMPLETE': ('Warning: )utstanding issues, refer to the RRA', 'done'),
```
Comment (claudijd) [inline casa.py]: I think you want " Outstanding" here.

File: assigner.py [hunk 3]
```diff
@@ -35,24 +37,119 @@ def main():
...
+        logger.critical("No CASA API Key passed in environment variable CASA_API_KEY")
+        sys.exit(127)
+    capi = casa.Casa(url=ccfg.get('url'), api_key=capi_key)
 
+    # Do things!
     autoassign(bapi, bcfg.get('rra'), args.dry_run)
     autoassign(bapi, bcfg.get('va'), args.dry_run)
+    autocasa(bapi, capi, bcfg, ccfg, args.dry_run)
 
+def autocasa(bapi, capi, bcfg, ccfg, dry_run):
```
Comment (claudijd) [inline assigner.py]: I wonder if it's better to de-couple CASA and generic BMO service assignment logic into two different scripts.  Basically, what I was suggesting here => https://github.com/mozilla/infosec-risk-management-bugzilla/issues/6

File: assigner.py (hunk 1 above)
Comment (gdestuynder) [reply assigner.py]: `XXX` is like `TODO` its a marker that shows up in many editor

File: casa.py (hunk 2 above)
Comment (gdestuynder) [reply casa.py]: woopsies

File: casa.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+            elif parse_next == 'creator':
+                parsed['creator'] = l.split('- ')[1]
+
+            # Setup next-line parser
+            if l == 'Biztera URL:':
+                parse_next = 'url'
+            elif l == 'Project Creator:':
+                parse_next = 'creator'
+            else:
+                parse_next = None
```
[the remark this answers is not in the document]
Comment (gdestuynder) [reply casa.py]: im not actually using the value, but left the parser in there since it may be useful at some point

File: assigner.py (hunk 3 above)
Comment (gdestuynder) [reply assigner.py]: i started with separate scripts, but since all configuration is shared and there is a depency issue i ended up integrating it

**Pull request merged**